sdk/src/main/scala/com/horizen/account/state/AccountState.scala

```diff
@@ -400,4 +359,109 @@ object AccountState {
     } else
       throw new RuntimeException("State metadata storage is not empty!")
   }
+
+  def blockGasLimitExceeded(cumGasUsed: BigInteger): Boolean = {
+    // TODO
+    false
+  }
+
+  def blockSizeExceeded(blockSize: Int, txCounter: Int): Boolean = {
+    if (txCounter > SidechainBlockBase.MAX_SIDECHAIN_TXS_NUMBER || blockSize > SidechainBlockBase.MAX_BLOCK_SIZE)
+      true // stop data collection
+    else {
+      false // continue data collection
+    }
+  }
+
+  def applyAndGetReceipts(stateView: AccountStateView,
```

Comment:
I honestly vote to separate the logic for state and forger.
As you may see this method is full of `if`s and there will be even more when forger will have more complex approach of transaction selection.
Let's keep the original code in the state as is, and have a separate consistent one for the Forger.

Reply:
Now using two separate methods

Reply:
Good.
I still tempted to avoid static methods here, but return it back to the AccountState class as it was before.
The same for the forger, put `blockGasLimitExceeded`, `blockSizeExceeded` and `tryApplyAndGetReceipts` methods to the Forger.

Reply:
Ok done, for the time being I just left the dummy `blockGasLimitExceeded` since it will be used by both modules